src/devices/blueline.c

```diff
@@ -0,0 +1,362 @@
+            /* clang-format off */
+            data = data_make(
+                    "model",         "", DATA_STRING, BLUELINE_MODEL,
+                    "id",            "", DATA_INT, context->current_sensor_id,
+                    "flags",         "", DATA_FORMAT, "%02x", DATA_INT, flags,
+                    "battery_ok",    "", DATA_INT,    !battery,
+                    "temperature_C", "", DATA_INT,    temperature,
+                    "mic",           "", DATA_STRING, "CRC",
+                    NULL);
+            /* clang-format on */
+            decoder_output_data(decoder, data);
+            payloads_decoded++;
+        } else { // Assume BLUELINE_ENERGY_MSG
+            // (The lowest two bits of the pulse count will always be the same because message_type is overlaid there)
+            const uint16_t pulses = offset_payload_u16;
+            /* clang-format off */
+            data = data_make(
+                    "model",    "", DATA_STRING, BLUELINE_MODEL,
+                    "id",       "", DATA_INT, context->current_sensor_id,
+                    "impulses", "", DATA_INT,    pulses,
+                    "mic",      "", DATA_STRING, "CRC",
+                    NULL);
+            /* clang-format on */
+            decoder_output_data(decoder, data);
+            payloads_decoded++;
+        }
+    }
+
+    return ((payloads_decoded > 0) ? payloads_decoded : most_applicable_failure);
+}
+
+static char *output_fields[] = {
+        "model",
+        "id",
+        "flags",
+        "gap",
+        "impulses",
+        "battery_ok",
+        "temperature_C",
+        "mic",
+        NULL,
+};
+
+r_device blueline;
+
+static r_device *blueline_create(char *arg)
+{
+    r_device *r_dev = create_device(&blueline);
+    if (!r_dev) {
+        fprintf(stderr, "blueline_create() failed\n");
+        return NULL; // NOTE: returns NULL on alloc failure.
+    }
+
+    struct blueline_stateful_context *context = malloc(sizeof(*context));
+    if (!context) {
+        WARN_MALLOC("blueline_create()");
+        free(r_dev);
+        return NULL; // NOTE: returns NULL on alloc failure.
+    }
+    memset(context,0,sizeof(*context));
+    r_dev->decode_ctx = context;
+
+    if (arg != NULL) {
+        if (strcmp(arg, "auto") == 0) {
+            // Setup for auto identification
+            context->searching_for_new_id = 1;
+            fprintf(stderr, "Blueline decoder will try to autodetect ID.\n");
+        } else {
+            // Assume user is trying to pass in hex ID
+            context->current_sensor_id = strtoul(arg, NULL, 0);
+            fprintf(stderr, "Blueline decoder using ID %u\n", context->current_sensor_id);
+        }
+    }
+
+    return r_dev;
+}
+
+r_device blueline = {
+        .name        = "BlueLine Power Monitor",
+        .modulation  = OOK_PULSE_PPM,
+        .short_width = 500,
+        .long_width  = 1000,
+        .gap_limit   = 2000,
+        .reset_limit = 200000,
```

Comment:
I think the practical limit for this is much smaller.

Reply:
Maybe 8000 would work?

Reply:
It's almost a don't-care then, as long as it's bigger than gap_limit.  The pauses between rows in one burst is actually closer to 100000, so they always show up in the decoder one row at a time.  I'll bring this down to 8000.